### PR TITLE
add musicbrainz api

### PR DESCRIPTION
**Pull request by paulwalko**
@echarlie

### CODE REVIEW

File: modules/musicbrainz.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
```
Comment (echarlie) [inline modules/musicbrainz.py:1]: python3

File: modules/musicbrainz.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# coding=utf-8
+"""
+musicbrainz.py - MusicBrainz API lookup
+Paul Walko
+"""
+
+from modules.search import musicbrainz_api
+
+def mz(phenny, input):
```
Comment (echarlie) [inline modules/musicbrainz.py:10]: .mb or .mbz instead.

File: modules/test/test_musicbrainz.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    def setUp(self):
+        self.phenny = MagicMock()
+
+    def test_mz_area(self):
+        input = Mock(group=lambda x: "area Blacksburg")
+        mz(self.phenny, input)
+
+        self.phenny.say.assert_called_once_with("Top area for \"Blacksburg\": https://musicbrainz.org/area/76d1e078-7861-4329-8553-f047b922c4f7")
+
+    def test_mz_artist(self):
```
Comment (echarlie) [inline modules/test/test_musicbrainz.py:22]: Use a better-known/less likely to be changed artist. e.g. the Beatles.

File: modules/musicbrainz.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    if result:
+        if "disambiguation" in result:
+            phenny.say("Top {} for \"{}\" ({}): https://musicbrainz.org/{}/{}".format(entity_type, query, result["disambiguation"], entity_type, result["id"]))
+        else:
+            phenny.say("Top {} for \"{}\": https://musicbrainz.org/{}/{}".format(entity_type, query, entity_type, result["id"]))
+    else:
+        phenny.reply("Sorry, no result.")
+mz.commands = ["mz"]
+mz.example = ".mz artist electrobro"
```
Comment (echarlie) [inline modules/musicbrainz.py:33]: example should use something well-known.

**Review by echarlie — CHANGES_REQUESTED** [review]

**Review by mutantmonkey — APPROVED** [review]

**Comment by paulwalko** [comment]
@echarlie

File: modules/test/test_musicbrainz.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+from mock import MagicMock, Mock
+from modules.musicbrainz import mb
+
+
+class TestMusicBrainz(unittest.TestCase):
+    def setUp(self):
+        self.phenny = MagicMock()
+
+    def test_mb_area(self):
+        input = Mock(group=lambda x: "area Blacksburg")
```
Comment (echarlie) [inline modules/test/test_musicbrainz.py:17]: probably not ideal, b/c Blacksburg, NC exists. But not blocking.

Comment (paulwalko) [reply modules/test/test_musicbrainz.py:17]: SC, actually

**Review by echarlie — APPROVED** [review]
I don't love the test cases, still. but approved.

File: modules/test/test_musicbrainz.py (hunk 5 above)
Comment (mutantmonkey) [reply modules/test/test_musicbrainz.py:17]: Chicago, New York, Portland, San Francisco, San Diego, or Seattle might be good alternatives

File: modules/test/test_musicbrainz.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        input = Mock(group=lambda x: "work 404")
+        mb(self.phenny, input)
+
+        self.phenny.say.assert_called_once_with("404, Song: https://musicbrainz.org/work/18a04f67-f105-42a5-9e6a-10bf349e5965")
+
+    def text_mb_recording(self):
+        input = Mock(group=lambda x: "recording basketball")
+        mb(self.phenny, input)
+
+        self.phenny.say.assert_called_once_with("Top recording for \"basketball\": https://musicbrainz.org/recording/b23393e7-3930-4687-b94b-970982d003e6")
```
Comment (paulwalko) [inline modules/test/test_musicbrainz.py:50]: will change

File: modules/test/test_musicbrainz.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        input = Mock(group=lambda x: "release-group hello")
+        mb(self.phenny, input)
+
+        self.phenny.say.assert_called_once_with("HELLO,HELLO,HELLO, Album: https://musicbrainz.org/release-group/ef20be97-cfb2-42f9-8a66-f650118c02e4")
+
+    def test_mb_work(self):
+        input = Mock(group=lambda x: "work 404")
+        mb(self.phenny, input)
+
+        self.phenny.say.assert_called_once_with("404, Song: https://musicbrainz.org/work/18a04f67-f105-42a5-9e6a-10bf349e5965")
```
Comment (paulwalko) [inline modules/test/test_musicbrainz.py:44]: will change

File: modules/test/test_musicbrainz.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        input = Mock(group=lambda x: "release The Dark Side Of The Moon")
+        mb(self.phenny, input)
+
+        self.phenny.say.assert_called_once_with("The Dark Side of the Moon: https://musicbrainz.org/release/90a5f2b9-e3ba-4742-8a35-e68c40f13839")
+
+    def test_mb_release_group(self):
+        input = Mock(group=lambda x: "release-group hello")
+        mb(self.phenny, input)
+
+        self.phenny.say.assert_called_once_with("HELLO,HELLO,HELLO, Album: https://musicbrainz.org/release-group/ef20be97-cfb2-42f9-8a66-f650118c02e4")
```
Comment (paulwalko) [inline modules/test/test_musicbrainz.py:38]: will change

File: modules/test/test_musicbrainz.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+class TestMusicBrainz(unittest.TestCase):
+    def setUp(self):
+        self.phenny = MagicMock()
+
+    def test_mb_area(self):
+        input = Mock(group=lambda x: "area Blacksburg")
+        mb(self.phenny, input)
+
+        self.phenny.say.assert_called_once_with("Blacksburg, City: https://musicbrainz.org/area/76d1e078-7861-4329-8553-f047b922c4f7")
```
Comment (paulwalko) [inline modules/test/test_musicbrainz.py:20]: will change

File: modules/test/test_musicbrainz.py [hunk 10]
```diff
@@ -29,22 +29,22 @@ def test_mb_release(self):
...
-        input = Mock(group=lambda x: "work 404")
+        input = Mock(group=lambda x: "work The Girl from Ipanema")
         mb(self.phenny, input)
 
-        self.phenny.say.assert_called_once_with("404, Song: https://musicbrainz.org/work/18a04f67-f105-42a5-9e6a-10bf349e5965")
+        self.phenny.say.assert_called_once_with("The Girl From Ipanema, Song: https://musicbrainz.org/work/69f81329-a815-3ec2-aaee-1f781a1e026d")
 
     def text_mb_recording(self):
-        input = Mock(group=lambda x: "recording basketball")
+        input = Mock(group=lambda x: "recording Never Gonna Give You Up")
```
Comment (mutantmonkey) [inline modules/test/test_musicbrainz.py:47]: :laughing:

**Review by mutantmonkey — APPROVED** [review]

**Comment by mutantmonkey** [comment]
Thanks; this is a great addition!

**Pull request merged**